mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -726,3 +752,21 @@ pub(super) struct SaplingSyncGuard<'a> {
     pub(super) _connector_guard: AsyncMutexGuard<'a, SaplingSyncConnector>,
     pub(super) respawn_guard: SaplingSyncRespawnGuard,
 }
+
+async fn send_multi_light_wallet_request<'a, Res, Fut, Fn>(
+    clients: &'a mut [CompactTxStreamerClient<Channel>],
+    mut req_fn: Fn,
+) -> Result<Res, Vec<tonic::Status>>
+where
+    Fut: Future<Output = Result<tonic::Response<Res>, tonic::Status>>,
+    Fn: FnMut(&'a mut CompactTxStreamerClient<Channel>) -> Fut,
+{
+    let mut errors = Vec::new();
```

Comment:
Refer to #1480, this flow isn't good solution for handling multiple http connections. In scenario where  we have 4-5 urls and first 3 always times out, the process will be unacceptably long.. There are similar implementation exists which will to be refactored as well.


Could you please add TODO note top of the this function including the issue url? We can refactor all of them in different PR.

Reply:
good note!
done